### PR TITLE
Add new GEMM instances with IGLP optimized pipeline v1 (IGLPStrategy=0)

### PR DESCRIPTION
**Pull request by poyenc**
I use _test_gemm_standalone_xdl_fp16_ as benchmark to collect performance number on an 104 CU machine (1 wave per SIMD).

|Layout|M|N|K|Tile Size|Pipeline v1 (TFlops)|Pipeline v1 - IGLP optimized (TFlops)|Improv (%)|
|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|
|_NN_|2048|3328|4096|256x256|6.32|16.13|**155.05**|
|_NN_|2048|1664|4096|256x128|98.62|96.87|-1.77|
|_NN_|1024|1664|4096|128x128|73.58|72.93|-0.89|
|_NN_|1024|832|4096|128x64|56.80|59.77|**5.23**|
|_NT_|2048|3328|4096|256x256|5.89|16.78|**184.96**|
|_NT_|2048|1664|4096|256x128|102.32|110.94|**8.42**|
|_NT_|1024|1664|4096|128x128|78.55|90.08|**14.67**|
|_NT_|1024|832|4096|128x64|55.73|66.22|**18.82**|
|_TN_|2048|3328|4096|256x256|6.86|17.15|**150.00**|
|_TN_|2048|1664|4096|256x128|103.45|103.49|0.04|
|_TN_|1024|1664|4096|128x128|78.19|73.13|-6.46|
|_TN_|1024|832|4096|128x64|53.82|50.79|-5.64|
|_TT_|2048|3328|4096|256x256|6.41|16.00|**149.61**|
|_TT_|2048|1664|4096|256x128|113.95|110.71|-2.84|
|_TT_|1024|1664|4096|128x128|79.43|83.33|4.90|
|_TT_|1024|832|4096|128x64|52.88|51.55|-2.52|

We can have some improvement by applying `__builtin_amdgcn_iglp_opt(0)` to the GEMM pipeline v1 (along with compilation options: **-mllvm -amdgpu-enable-max-ilp-scheduling-strategy**). In this PR, I had extended `GridwiseGemmPipeline_v1<>` with an extra parameter `IGLPOptStrategy` which enables users to choose _IGLPStrategy_ (the argument of `__builtin_amdgcn_iglp_opt()` built-in function, support future LLVM updates).

### CODE REVIEW

**Comment by poyenc** [comment]
Need to figure a proper  way to add compilation options **-mllvm -amdgpu-enable-max-ilp-scheduling-strategy** to reserve `__builtin_amdgcn_iglp_opt()` altered scheduling sequence

File: library/src/tensor_operation_instance/gpu/gemm/device_gemm_xdl_c_shuffle_f16_f16_f16_km_kn_mn_instance.cpp [hunk 1]
```diff
@@ -53,6 +53,26 @@ using device_gemm_xdl_c_shuffle_f16_f16_f16_km_kn_mn_instances = std::tuple<
...
+#if CK_EXPERIMENTAL_IGLP_OPT_INSTANCES
+        // pipeline v1, 1 wave (__builtin_amdgcn_iglp_opt() optimized)
+        ,
+        DeviceGemm_Xdl_CShuffle<     Col,      Row,    Row,   F16,   F16,   F16,     F32,      F16, PassThrough, PassThrough, PassThrough,    GemmDefault,        1,   256,   256,   128,    32,   2,   2,   32,   32,    4,    2,     S<4, 64, 1>,     S<0, 2, 1>,     S<0, 2, 1>,              1,              4,              2,         0,     S<8, 32, 1>,     S<0, 2, 1>,     S<0, 2, 1>,             1,              4,              2,         0,           1,           1,               S<1, 32, 1, 8>,              8,  LoopScheduler::Default,        PipelineVersion::v1_opt0>,
```
Comment (rosenrodt) [inline library/src/tensor_operation_instance/gpu/gemm/device_gemm_xdl_c_shuffle_f16_f16_f16_km_kn_mn_instance.cpp:59]: Should we templatize the pipeline? Something like the one below and instantiate them all at once
```
template <PipelineVersion pipeline> 
using device_gemm_xdl_c_shuffle_f16_f16_f16_km_kn_mn_instances = ...
```

Comment (poyenc) [reply library/src/tensor_operation_instance/gpu/gemm/device_gemm_xdl_c_shuffle_f16_f16_f16_km_kn_mn_instance.cpp:59]: Sure, we can use template to eliminate all the duplicated codes. But I'm not sure should I use `PipelineVersion::v1_opt0` to represent the optimized pipeline.

**Comment by poyenc** [comment]
as discussed, we will try to optimize pipeline v2 and compare result with Tensile

**Pull request closed without merging**